src/App/PropertyStandard.cpp

```diff
@@ -1359,21 +1359,21 @@ void PropertyString::setValue(const char* newLabel)
 
                 bool changed = false;
                 label = label.substr(0,lastpos+1);
-                if(label != obj->getNameInDocument()
+                std::string objName = obj->getNameInDocument();
+                if(label != objName
                         && boost::starts_with(obj->getNameInDocument(),label))
                 {
                     // In case the label has the same base name as object's
                     // internal name, use it as the label instead.
-                    const char *objName = obj->getNameInDocument();
-                    const char *c = &objName[lastpos+1];
+                    const char *c = &objName.c_str()[lastpos+1];

```

Comment:
Weird construct. Why not writing `const char c = objName[lastpos + 1];`? The code below needs to be adjusted to avoid this ugly pointer arithmetic.